src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -331,7 +341,7 @@ public static object CoerceValue(ISchema schema, IGraphType type, IValue input,
                     var objectField = objectValue.Field(field.Name);
                     if (objectField != null)
                     {
-                        obj[field.Name] = CoerceValue(schema, field.ResolvedType, objectField.Value, variables) ?? field.DefaultValue;
+                        obj[field.Name] = CoerceValue(schema, field.ResolvedType, objectField.Value, variables);
```

Comment:
```suggestion
                        obj[field.Name] = CoerceValue(schema, field.ResolvedType, objectField.Value, variables, field.DefaultValue);
```

?

If so, then we do not have the required test.

Reply:
Not in this case.  I looked at this a bunch of times also to be sure.  But if the field exists in the object array (`objectField != null`), then we want to capture it's value, even if it's null.  So the default value does not apply.

Reply:
A few lines below you can see if the field was not specified, and if a default value was specified, then add it to the dictionary (aka map).

Reply:
The behavior according to the spec is exceedingly specific here:

> If no value is provided for a defined input object field and that field definition provides a default value, the default value should be used. If no default value is provided and the input object field’s type is non‐null, an error should be thrown. Otherwise, if the field is not required, then no entry is added to the coerced unordered map.

So as you can see we are required to not add a null value to the dictionary if there is no dictionary and no default value, etc.

Reply:
But if it's null:

> If the value null was provided for an input object field, and the field’s type is not a non‐null type, an entry in the coerced unordered map is given the value null. In other words, there is a semantic difference between the explicitly provided value null versus having not provided a value.

Reply:
> But if the field exists in the object array (objectField != null), then we want to capture it's value, even if it's null. So the default value does not apply.

Better to add in comments.

Reply:
Also see this https://github.com/graphql/graphql-spec/pull/793

Reply:
I think I didn't cover this rule, so I'll make a test and fix it:

> If no default value is provided and the input object field’s type is non‐null, an error should be thrown.

Reply:
`inputRequired` field in test seems to fit to such a test

Reply:
It's covered by `AssertValidVariableValue`

Reply:
Comments added.

Reply:
OK